### PR TITLE
ci-checks: run all the checks on checked-in files; check shellcheck

### PR DESCRIPTION
**Pull request by grahamc**
Don't run on files which are not being committed, and run all
the checks even if one step failed.

This PR should fail (validating that the CI check still fails) and then I will add additional commits fixing its complaints.

### CODE REVIEW

**Comment by grahamc** [comment]
@mmlb do you think it is worth adding `git` to the shell.nix for this, or should we go back to using wildcards? I wanted to use `git ls-files` so that untracked files in a checkout don't gum up the works.

**Comment by mmlb** [comment]
I'm fine with adding git to shell.nix. The shell.nix's aren't super pure for the projects and I've kind of wanted to move towards that way as much as possible where it makes sense.

File: test/push_images.sh [hunk 1]
```diff
@@ -1,10 +1,6 @@
 #!/bin/bash
 
-for i in {1..10}; do
-	docker login -u username -p password localhost
-	if [ $? -eq 0 ]; then
-		break
-	fi
+while ! docker login -u username -p password localhost; do
```
Comment (mmlb) [inline test/push_images.sh:3]: Does this really belong in this PR? I'm fine if you want to earmark it in, just wand to double check.

Comment (grahamc) [reply test/push_images.sh:3]: Maybe not, it just seemed like the easiest fix for `$i` being unused. I could swap it out to just echo the iteration. What do you think?

**Pull request merged**